src/Control/Lens/Traversal.hs

```diff
@@ -1466,3 +1468,28 @@ traverseByOf l pur app f = reifyApplicative pur app (l (ReflectedApplicative #.
 -- @
 sequenceByOf :: Traversal s t (f b) b -> (forall x. x -> f x) -> (forall x y. f (x -> y) -> f x -> f y) -> s -> f t
 sequenceByOf l pur app = reifyApplicative pur app (l ReflectedApplicative)
+
+-- | A version of 'Control.Lens.Setter.over' that forces the result(s) of
+-- applying the function. This can prevent space leaks when modifying lazy
+-- fields.
+--
+-- @
+-- over traverse (const ⊥) [1,2] = [⊥, ⊥]
```

Comment:
These can be turned into proper doctests. `length $ ...`  and matching like

```
...
...Exception...
...
```

for an error. It's not the prettiest, but  it's an honest test.

Reply:
To my mind doctests wouldn't do as good a job of documenting, but I can write them anyway. How do I run those locally?

Reply:
```
cabal build all --enable-tests
cabal-docspec
```

Reply:
> To my mind doctests wouldn't do as good a job of documenting, but I can write them anyway.

There is barely non-doctest examples in `lens`. They may do less good job as documenting, but at least they stay up to date. There are *hundreds* of examples in `lens` package. No-one would ever go through them by hand and verify they still type-check/work/etc.

Reply:
I'm getting errors from `cabal-doctest` about mismatched GHC and plan versions. Unfortunately, `cabal-doctest` doesn't seem to have much in the way of documentation, so I can't figure out where to go from there.